apis/autoscaling/v1alpha1/mariadb_types.go

```diff
@@ -17,172 +17,96 @@ limitations under the License.
-	// If not set, the default value is to allow to scale down to minReplicas pods, with a
-	// 300 second stabilization window (i.e., the highest recommendation for
-	// the last 300sec is used).
-	// +optional
-	ScaleDown *MariaDBScalingRules `json:"scaleDown,omitempty" protobuf:"bytes,2,opt,name=scaleDown"`
-}
-
-// MariaDBScalingRules configures the scaling behavior for one direction.
-// These Rules are applied after calculating DesiredReplicas from metrics for the HPA.
-// They can limit the scaling velocity by specifying scaling policies.
-// They can prevent flapping by specifying the stabilization window, so that the
-// number of replicas is not set instantly, instead, the safest value from the stabilization
-// window is chosen.
-type MariaDBScalingRules struct {
-	// StabilizationWindowSeconds is the number of seconds for which past recommendations should be
-	// considered while scaling up or scaling down.
-	// StabilizationWindowSeconds must be greater than or equal to zero and less than or equal to 3600 (one hour).
-	// If not set, use the default values:
-	// - For scale up: 0 (i.e. no stabilization is done).
-	// - For scale down: 300 (i.e. the stabilization window is 300 seconds long).
-	// +optional
-	StabilizationWindowSeconds *int32 `json:"stabilizationWindowSeconds" protobuf:"varint,3,opt,name=stabilizationWindowSeconds"`
-	// selectPolicy is used to specify which policy should be used.
-	// If not set, the default value MaxPolicySelect is used.
-	// +optional
-	SelectPolicy *v2beta2.ScalingPolicySelect `json:"selectPolicy,omitempty" protobuf:"bytes,1,opt,name=selectPolicy"`
-	// policies is a list of potential scaling polices which can be used during scaling.
-	// At least one policy must be specified, otherwise the MariaDBScalingRules will be discarded as invalid
-	// +optional
-	Policies []MariaDBScalingPolicy `json:"policies,omitempty" protobuf:"bytes,2,rep,name=policies"`
+type MariaDBComputeAutoscalerSpec struct {
+	MariaDB          *ComputeAutoscalerSpec `json:"mariadb,omitempty" protobuf:"bytes,1,opt,name=mariadb"`
+	DisableScaleDown bool                   `json:"disableScaleDown,omitempty" protobuf:"varint,6,opt,name=disableScaleDown"`
 }
 
-// MariaDBScalingPolicy is a single policy which must hold true for a specified past interval.
-type MariaDBScalingPolicy struct {
-	// Type is used to specify the scaling policy.
-	Type v2beta2.HPAScalingPolicyType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=HPAScalingPolicyType"`
-	// Value contains the amount of change which is permitted by the policy.
-	// It must be greater than zero
-	Value int32 `json:"value" protobuf:"varint,2,opt,name=value"`
-	// PeriodSeconds specifies the window of time for which the policy should hold true.
-	// PeriodSeconds must be greater than zero and less than or equal to 1800 (30 min).
-	PeriodSeconds int32 `json:"periodSeconds" protobuf:"varint,3,opt,name=periodSeconds"`
+type MariaDBStorageAutoscalerSpec struct {
```

Comment:
Why do we need `mariadb` under `storage` section when it is `MariaDBAutoscaler`? What else cloud be in `storage`?
```
spec:
  storage:
    mariadb:
      .. .. .. 
```
Why don't you use it?

```yaml
type MariaDBAutoscalerSpec struct {
  DatabaseRef *core.LocalObjectReference 
  Compute *MariaDBComputeAutoscalerSpec 
  Storage *MariaDBStorageAutoscalerSpec 
}

type MariaDBStorageAutoscalerSpec struct {
       StorageAutoscalerSpec
}
```